java/util/src/main/java/com/google/protobuf/util/JsonFormat.java

```diff
@@ -151,12 +155,22 @@ public Printer includingDefaultValueFields() {
      * current {@link Printer}.
      */
     public Printer preservingProtoFieldNames() {
-      return new Printer(registry, includingDefaultValueFields, true);
+      return new Printer(registry, includingDefaultValueFields, true, omittingInsignificantWhitespace);
+    }
+
+
+    /**
+     * Creates a new {@link Printer} that has a different textGenerator, The json output will be omittingInsignificantWhitespace.
```

Comment:
=> Create a new Printer that will omit all insignificant whitespace in the JSON output. This new Printer clones all other configurations from the current Printer. Insignificant whitespace is defined by the JSON spec as whitespace that appear between JSON structural elements:
<pre>

      ws = *(
              %x20 /              ; Space
              %x09 /              ; Horizontal tab
              %x0A /              ; Line feed or New line
              %x0D )              ; Carriage return
</pre>
See https://tools.ietf.org/html/rfc7159